opengate/actors/filters.py

```diff
@@ -23,9 +23,21 @@ class FilterBase(GateObject):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
+    def __initcpp__(self):
+        """Nothing to do in the base class."""
+
     def initialize(self):
         self.InitializeUserInput(self.user_info)
 
+    def __getstate__(self):
+        state_dict = super().__getstate__()
+        state_dict["filter_engine"] = None
```

Comment:
Why do we need to set this to None? I have the impression that the filter_engine attribute is actually never set, so there should not be any need for setting it to None. I think the __getstate__ method is not needed. The __setstate__ method yes.